src/lib/include/vector_triple.h

```diff
@@ -85,10 +94,19 @@ template <typename VectorT> class VectorTriple {
       : x(a), y(b), z(c) {}
 
   // construct by loading from an array of ScalarT eg float* or double *.
-  inline VectorTriple(const ScalarT *source)
-      : x(_genericload<VectorT>(source)),
-        y(_genericload<VectorT>(&source[ValuesPerPack<VectorT>])),
-        z(_genericload<VectorT>(&source[+2 * ValuesPerPack<VectorT>])) {}
+  inline VectorTriple(const ScalarT *source) {
+    if (ValuesPerPack<VectorT> == 1) {
```

Comment:
I wish this could be `constexpr` but alas it cannot be if we do not require C++ 17.